pkg/util/intstr/intstr.go

```diff
@@ -113,3 +115,29 @@ func (intstr *IntOrString) Fuzz(c fuzz.Continue) {
 		c.Fuzz(&intstr.StrVal)
 	}
 }
+
+func GetValueFromIntOrPercent(intOrStr *IntOrString, value int) (int, error) {
+	maxSurge, isPercent, err := getIntOrPercentValue(intOrStr)
```

Comment:
This isn't necessarily maxSurge. The same pattern applies to maxAvailable.

Besides, this is a generic utility.

I would rename `intOrString` to `intOrPercent`, `value` to `total`, and `maxSurge` to `value`.